util/topgen/c_test.py

```diff
@@ -59,13 +59,15 @@ def __init__(self, name: str, inst_name: str, base_addr_name: str,
 
 
 class TopGenCTest(TopGenC):
-    def __init__(self, top_info, name_to_block: Dict[str, IpBlock]):
-        super().__init__(top_info, name_to_block)
+    def __init__(self, top_info, name_to_block: Dict[str, IpBlock], addr_space: str):
+        super().__init__(top_info, name_to_block, addr_space)
 
         self.irq_peripherals = self._get_irq_peripherals()
         self.alert_peripherals = self._get_alert_peripherals()
 
     def _get_irq_peripherals(self):
+        if self.addr_space != self.default_addr_space:
+            return []
```

Comment:
I'm not totally sure whether this restriction would generally hold, but it doesn't hurt to start with it.

Reply:
No, in the future, I would like to determine if there is a PLIC in the address space and then yield the devices that are connected to that one - keyword `interrupt_domain`.